tools/python/clean_strings_txt.py

```diff
@@ -0,0 +1,323 @@
+
+    return parser.parse_args()
+
+
+def do_multiple(args):
+    write_filtered_strings_txt(
+        grep_ios(), "{0}/ios_strings.txt".format(OMIM_ROOT), args.langs
+    )
+    write_filtered_strings_txt(
+        grep_android(), "{0}/android_strings.txt".format(OMIM_ROOT), args.langs
+    )
+    if args.generate:
+        print("Going to generate locs")
+        exec_shell(
+            "{0}/tools/unix/generate_localizations.sh".format(OMIM_ROOT),
+            "android_strings.txt ios_strings.txt"
+        )
+
+
+def generate_auto_tags(ios, android):
+    new_tags = defaultdict(list)
+    for i in ios:
+        new_tags[i].append("ios")
+
+    for a in android:
+        new_tags[a].append("android")
+
+    for key in new_tags:
+        new_tags[key] = ", ".join(new_tags[key])
+
+    return new_tags
+
+
+def new_comments_and_tags(strings_txt, filtered, new_tags):
+    comments_and_tags = {key: strings_txt.comments_and_tags[key] for key in filtered}
+    for key in comments_and_tags:
+        comments_and_tags[key]["tags"] = new_tags[key]
+    return comments_and_tags
+
+
+def do_single(args):
+    ios = grep_ios()
+    android = grep_android()
+
+    new_tags = generate_auto_tags(ios, android)
+
+    filtered = ios
+    filtered.update(android)
+
+    strings_txt = StringsTxt()
+    strings_txt.translations = {key: dict(strings_txt.translations[key]) for key in filtered}
+
+    strings_txt.comments_and_tags = new_comments_and_tags(strings_txt, filtered, new_tags)
+
+    path = args.output if isabs(args.output) else "{0}/{1}".format(OMIM_ROOT, args.output)
+    strings_txt.write_formatted(languages=args.langs, target_file=path)
+
+    if args.generate:
+        exec_shell(
+            "{}/../unix/generate_localizations.sh".format(OMIM_ROOT),
+            "{0} {0}".format(args.output)
+        )
+
+def do_missing(args):
+    ios = set(grep_ios())
+    strings_txt_keys = set(StringsTxt().translations.keys())
+    missing = ios - strings_txt_keys
+
+    if missing:
+        for m in missing:
+            logging.info(m)
+        exit(1)
+    else:
+        logging.info("Ok. No missing strings.")
+        exit(0)
+
+
+def do_candidates(args):
+    all_candidates = defaultdict(list)
+    for source, candidate in grep_ios_candidates():
+        all_candidates[candidate].append(source)
+
+    for candidate, sources in all_candidates.iteritems():
+        print(candidate, sources)
+
+
+def do_ios_suspects(args):
+    grep = "grep -re -I 'L(' {}/iphone/*".format(OMIM_ROOT)
+    suspects = exec_shell(grep)
+    SUSPECT_RE = re.compile(r"(.*?):.*?\WL\(([^@].*?)\)")
+    strings = strings_from_grepped(suspects, SUSPECT_RE)
+    for s in strings:
+        print(s)
+
+
+def find_omim():
+    my_path = abspath(__file__)
+    tools_index = my_path.rfind("/tools/python")
+    omim_path = my_path[:tools_index]
+    return omim_path
+
+
+def read_hardcoded_categories(a_path):
+    with open(a_path) as infile:
+        return [s.strip() for s in infile if s]
```

Comment:
What if s is something like "   ", i.e. a string of spaces? In this case you'll return it here, as empty string.